src/Engine/Component/ParticleComponents/ParticleComponent.cpp

```diff
@@ -0,0 +1,306 @@
+            Vector<Mesh *> *meshes = getMeshes(effect);
+            Vector<ModelTexture*> * textures = getTextures(effect);
+            return ParticleEffect::createEffectParams(type, n, effectDuration, particleDuration, orientations,
+                randomDistribution, variance, rate, angle, loop, magnitude, attenuation, accelerators, meshes, textures);
+        }
+        default:
+            return NULL;
+    }
+        
+}
+
+Vector<glm::vec3> * ParticleComponent::getParticlePositions() {
+    return m_activeEffect->ActiveParticlePositions();
+}
+
+Vector<int> * ParticleComponent::getParticleOrientationIDs() {
+    return m_activeEffect->ActiveParticleOrientationIDs();
+}
+
+void ParticleComponent::initRandomOrientations(glm::mat4 comp, glm::mat3 norm, int count) {
+    m_randomMs = Vector<glm::mat4>();
+    m_randomMs.push_back(comp);
+    m_randomNs = Vector<glm::mat3>();
+    m_randomNs.push_back(norm);
+    for (int i = 0; i < count; i++) {
+        float angle = 2 * glm::pi<float>() * static_cast <float> (rand()) / static_cast <float> (RAND_MAX);
+        glm::fvec3 & axis = glm::ballRand(1.0f);
+
+        const glm::fmat4 & rot = glm::rotate(angle, axis);
+        const glm::fvec3 & translation = glm::fvec3(0.0f);
+        const glm::fvec3 & scale = glm::fvec3(getScaleFactor(m_effect));
+        m_randomNs.emplace_back(glm::mat3(rot) * glm::mat3(glm::scale(glm::mat4(), 1.0f / scale)));
+        m_randomMs.emplace_back(Util::compositeTransform(scale, rot, translation));
+    }
+}
+
+Mesh* ParticleComponent::getMesh(int i) {
+    return m_activeEffect->getMesh(i);
+}
+
+ModelTexture* ParticleComponent::getModelTexture(int i) {
+    return m_activeEffect->getModelTexture(i);
+}
+
+Vector<Mesh*> * ParticleComponent::getMeshes(ParticleEffect::Effect effect) {
+    Vector<Mesh*> * meshes = new Vector<Mesh*>();
+    switch (effect) {
+        case ParticleEffect::Effect::BLOOD_SPRAY:
+        {
+            meshes->push_back(Loader::getMesh("particles/Blood_Drop.obj"));
+            return meshes;
+        }
+        case ParticleEffect::Effect::WATER_FOUNTAIN:
+        {
+            meshes->push_back(Loader::getMesh("particles/Water_Drop.obj"));
+            return meshes;
+        }
+        case ParticleEffect::Effect::SODA_GRENADE:
+        {
+            meshes->push_back(Loader::getMesh("Hamburger.obj"));
+            return meshes;
+        }
+        case ParticleEffect::Effect::BODY_EXPLOSION:
+        {
+            meshes->push_back(Loader::getMesh("particles/Gore_Chunk.obj"));
+            return meshes;
+        }
+        case ParticleEffect::Effect::SIRACHA_FLAMETHROWER:
+        {
+            meshes->push_back(Loader::getMesh("particles/Blood_Drop.obj"));
+            return meshes;
+        }
+        default:
+            return NULL;
+    }
+}
+
+Vector<ModelTexture*> * ParticleComponent::getTextures(ParticleEffect::Effect effect) {
+    Vector<ModelTexture*> * textures = new Vector<ModelTexture*>();
```

Comment:
Is this new call being cleaned up anywhere?

Reply:
Now in particle System.